pkg/controllers/kubeadminpwd/oauthrestarter.go

```diff
@@ -1,77 +1,93 @@
 package kubeadminpwd
 
 import (
-	"context"
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
 
 	"github.com/go-logr/logr"
 
-	appsv1 "k8s.io/api/apps/v1"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclient "k8s.io/client-go/kubernetes"
+	corelisters "k8s.io/client-go/listers/core/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
-	OAuthDeploymentName = "oauth-openshift"
+	OAuthDeploymentName  = "oauth-openshift"
+	SecretHashAnnotation = "hypershift.openshift.io/kubeadmin-secret-hash"
 )
 
 type OAuthRestarter struct {
-	// Client is a client of the management cluster
-	client.Client
+	// Client is a client that allows access to the management cluster
+	Client kubeclient.Interface
 
 	// Log is the logger for this controller
 	Log logr.Logger
 
 	// Namespace is the namespace where the control plane of the cluster
 	// lives on the management server
 	Namespace string
+
+	// SecretLister is a lister for target cluster secrets
+	SecretLister corelisters.SecretLister
 }
 
 func (o *OAuthRestarter) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	controllerLog := o.Log.WithValues("pod", req.NamespacedName.String())
-	ctx := context.Background()
-
-	// Ignore any namespace that is not the Namespace.
-	if req.Namespace != o.Namespace {
-		return ctrl.Result{}, nil
-	}
+	controllerLog := o.Log.WithValues("secret", req.NamespacedName.String())
 
-	// Ignore all pods except the manifest bootstrapper
-	if req.Name != ManifestBootstrapperPod {
+	// Ignore any secret that is not kube-system/kubeadmin
+	if req.Namespace != metav1.NamespaceSystem || req.Name != KubeAdminSecret {
```

Comment:
should be `&&` as we don't want secret kube-system/ignore-this-secret to get through

Reply:
nevermind this comment. i shouldn't be reviewing PRs this late on a friday :)